### DISCUSSION TITLE
Field.Root: Rules of hooks violation

### DISCUSSION THREAD

**Discussion by jonathan-sawmills** [chakra-ui/ark#2645] [General]
Hello! I've encountered a React rules of hooks violation inside Field's `useField` hook.

Rules of hooks: https://react.dev/reference/rules/rules-of-hooks#only-call-hooks-at-the-top-level
> Don’t call Hooks inside loops, conditions, nested functions, or try/catch/finally blocks. Instead, always use Hooks at the top level of your React function, before any early returns. You can only call Hooks while React is rendering a function component
> 🔴 Do not call Hooks inside conditions or loops.

The violation is here (`useId` is conditionally called):
https://github.com/chakra-ui/ark/blob/main/packages/react/src/components/field/use-field.ts#L33

**Comment by segunadebayo** [comment]
Thanks for fix @jonathan-sawmills
